src/doc-editors/ServerGroupsEditor.vue

```diff
@@ -560,6 +571,27 @@ export default defineComponent({
     },
   },
   methods: {
+    async generateCertificate(certificateId?: string) {
+      this.isDownloadLoading = true
+      const serverGroupText = this.titles['certificate-singular']
+      const successMessage = `The ${serverGroupText} was deleted.`
+      const failureMessage = `Failed while attempting to delete the ${serverGroupText}.`
+      const method = 'POST'
+      if (certificateId) {
+        const url = `configs/${this.selectedBranch}/d/certificates/e/${certificateId}?domains=${this.selectedServerGroup.server_names}&site-id=${this.selectedServerGroup.id}`
+        await RequestsUtils.sendReblazeRequest(
+          {methodName: method,
+            url: url,
+            successMessage,
+            failureMessage,
+          })
+      } else {
+        const url = `configs/${this.selectedBranch}/d/certificates/e/${DatasetsUtils.generateUUID2()}?domains=${this.selectedServerGroup.server_names}&site-id=${this.selectedServerGroup.id}`
+        await RequestsUtils.sendReblazeRequest({methodName: method, url: url})
+      }
```

Comment:
Can we do something like that?
Trying to reduce the duplicated code
Indents may need fixing from the suggestion

```suggestion
      if (!certificateId) {
        certificateId = DatasetsUtils.generateUUID2()
      }
      const url = `configs/${this.selectedBranch}/d/certificates/e/${certificateId}?domains=${this.selectedServerGroup.server_names}&site-id=${this.selectedServerGroup.id}`
      await RequestsUtils.sendReblazeRequest(
        {methodName: method,
          url: url,
          successMessage,
          failureMessage,
        })
```

Reply:
Done